tests/test_io.py

```diff
@@ -1931,51 +1932,77 @@ def _test_result_exception(self, result_type, result_file_name="", fmi_type="me"
             with nose.tools.assert_raises(ResultSizeError):
                 res = model.simulate(options=opts)
 
-        def _test_result_size_verification(self, result_type, result_file_name=""):
+        def _test_result_size_verification(self, result_type, result_file_name="", dynamic_diagnostics=False):
             model, opts = self._setup(result_type, result_file_name)
+            model.setup_experiment()
+            model.initialize()
 
             max_size = 1e6
             opts["result_max_size"] = max_size
-            opts["ncp"] = 10000
+            opts["dynamic_diagnostics"] = dynamic_diagnostics
+            opts["logging"] = dynamic_diagnostics
+            opts["ncp"] = 0 #Set to zero to circumvent the early size check
+            ncp = 10000
+
+            result_handler = get_result_handler(model, opts)
+
+            result_handler.set_options(opts)
+            result_handler.initialize_complete()
+
+            if opts["dynamic_diagnostics"]:
+                opts['CVode_options']['rtol'] = 1e-6
+                opts['CVode_options']['atol'] = model.nominal_continuous_states * opts['CVode_options']['rtol']
+                diag_params, diag_vars = setup_diagnostics_variables(model, 0, opts, opts['CVode_options'])
+                result_handler.simulation_start(diag_params, diag_vars)
+            else:
+                result_handler.simulation_start()
 
             with nose.tools.assert_raises(ResultSizeError):
-                res = model.simulate(options=opts)
+                for i in range(ncp):
```

Comment:
Since we now have 2 distinct `ResultSizeError` sources, I think we should check that we do in fact raise/catch the intended one when testing.

Reply:
This is tested in such a way that the result file is verified to be within bounds. I'm hesitant to do a check on the actual Exception message.

Reply:
Right, seems OK then!